ast/verifier/Verifier.cc

```diff
@@ -16,6 +18,23 @@ class VerifierWalker {
         return original;
     }
 
+    unique_ptr<MethodDef> preTransformMethodDef(core::Context ctx, unique_ptr<MethodDef> original) {
+        methodDepth++;
+        return original;
+    }
+
+    unique_ptr<Expression> postTransformMethodDef(core::Context ctx, unique_ptr<MethodDef> original) {
+        methodDepth--;
+        return original;
+    }
+
+    unique_ptr<Expression> postTransformAssign(core::Context ctx, unique_ptr<Assign> original) {
+        if (ast::isa_tree<ast::UnresolvedConstantLit>(original->lhs.get())) {
+            ENFORCE(methodDepth == 0, "Found constant definition inside method definition");
+        }
+        return original;
+    }
+
```

Comment:
This adds a check to the verified that verifies that no constant assignments exist inside method defs.